### ISSUE TITLE
NPCs digging thru walls

### ISSUE THREAD

**Issue by AquariusPower** [Attnam/ivan#467]
with the ability to build (craft PR) walls,
we get completely safe by just digging, build a wall and remain there to rest and craft things...

I wonder if NPCs (mobs) that have any clue about us (ESP, sounds we make etc) could try to kick  (golems) or dig (dwarves or any other humanoid with a pickaxe) near us, not directly to us so we have a chance to remain hidden by keeping quiet for some time :)

I am not sure NPC can apply pick-axes, I followed the code once but remained uncertain about it...
as walls have "hit points" may be a simpler (than digging) calculation could be used, just kicking it would do also.

**Comment by fejoa** [comment]
There's always Vlad.

**Comment by AquariusPower** [comment]
haven't seen other than in the gfx files but I guess the huge aranea is also a clue on how to implement it :)
I just won a fight against enner beasts and I am still wondering about that portal while I dismantled 8 wooden chests to build a large one xD

**Comment by red-kangaroo** [comment]
Also mages are more deadly if you surround yourself with walls and cannot move.

**Comment by AquariusPower** [comment]
I think necromancers seem to target us with shock bolts w/o directional aiming, so if they can see us we are fried. 
They can also teleport, would be fun they aim teleport near were we are, despite would be quite non tactical for them :).

but I guess you are talking about mages that can damage without a walls' free linesight (so, damage thru walls) right?
haven't played against those yet.

I think the tactic would be to dig a big tunnel, far away from where they can detect us. Add a wall in the first dig spot and another in the last one, then we can safely craft and rest.

I am trying to let a missing feature properly work: secret doors. So mobs would not be able to open them so easily. May be the fake wall quality could be based on our perception, and if the mob has higher they would be able to spot it and kick or try to open them. https://github.com/Attnam/ivan/issues/468

But it seems, mobs may be spawned anywhere far from player (so still within that tunnel),
I wonder if adding light to squares could prevent their spawning near the lights,
but that may be a cheap (despite fun) trick to create a as safe as possible player's HQ inside a dungeon :)

**Comment by red-kangaroo** [comment]
@AquariusPower 
> but I guess you are talking about mages that can damage without a walls' free linesight (so, damage thru walls) right?

Yeah, mages can target you through ESP, so as long as you are in range, no matter the walls, they will blast you. It's fun. :P

**Comment by AquariusPower** [comment]
can't wait to find them! xD

**Comment by RustKnight** [comment]
@AquariusPower Sorry for the offtopic, but could you write me on this e-mail ? 
strandedfox33@gmail.com
I wanted to ask you something IVAN-ish related. 

Thanks!

**Comment by AquariusPower** [comment]
@RustKnight I dont know that much about Ivan, never completed the game, you will probably get a better feedback creating a new issue :)

**Comment by fejoa** [comment]
Hey @RustKnight! Are you still on the Attnam.com forums?

**Comment by RustKnight** [comment]
Hey - actually found my account - and yes I managed to hack my own password. So I am now active again (wow, 2009 , it's been some time)

**Comment by RustKnight** [comment]
@AquariusPower  Actually - it was coding related and I see you did some contribution yourself. If you have time to set a greenhorn programmer on the right tracks, I'd be glad to hear some advice.

**Comment by AquariusPower** [comment]
@RustKnight 

well, to create all stuff I did I ended learning a lot where to code things.

I think the best way is to make a question like: 
"How could I implement this thing?" and I may probably know where you could code it and how, even if it take some day to figure out :)

Anyway, the best place to ask it is the issues section so it is visible to everyone to contribute, learn and teach (me) back if it would be better coded differently xD

Another good thing is from these talks (and any other talks), create documents inside the project (as there are many already at Docs folder, some already copied and adjusted from posts here) to store the most important dev info being talked on github and Attnam.com, things that may help new developers to quick start doing things more easily.
